.github/workflows/publish_docker.yml

```diff
@@ -0,0 +1,163 @@
+name: Build Production
+on:
+  push:
+    branches:
+      - "main"
+jobs:
+  release:
+    runs-on: ubuntu-20.04
+    outputs:
+      published: ${{ steps.semantic.outputs.new_release_published }}
+      version: ${{ steps.semantic.outputs.new_release_version }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: semantic
+        uses: cycjimmy/semantic-release-action@v3
+        with:
+          semantic_version: 18
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN_PROJECT_ACTION }}
+
+  docker_x86_release:
```

Comment:
https://github.com/supabase/postgres/blob/develop/.github/workflows/dockerhub-release.yml

Can we follow this new example using build matrix? It reduces quite a bit of duplication.